### ISSUE TITLE
Apple M-series Support?

### ISSUE THREAD

**Issue by mrkai77** [WiiController/WiiController#9]
I recently got a M1 MBA but it turns out this app isn't compatible with Apple M-series CPUs.
I was wondering if the WiiController developers are working a ARM version of this app?

Any info is highly appreciated. Thanks!

**Comment by ThatsJustCheesy** [comment]
Hi there! While I would love to work with you to create an ARM-compatible version, Apple's hardening of kernel extensions in macOS 11 and on M1 hardware precludes a simple fix.

If #7 gets resolved, then I certainly wouldn't mind working on this—the actual kernel-resident code here is absolutely minimal (the original design by alxn1 is nice in this regard). However, to enable it you'd have to follow the instructions in the row labelled "Not enrolled" [here](https://support.apple.com/en-ca/guide/deployment-reference-macos/apd37565d329/web).

As I've said in #7, the correct long-term solution is to reimplement this as a "dext" which runs in user space. This would require 1) a time investment and 2) dropping macOS 10.14 and below. Although in theory, the kext and dext could be maintained in parallel, so we could eliminate problem (2) if we inflate problem (1) further.

So essentially, when I find time and motivation, I'd love to make this work on macOS 11 / M-series.

In the meantime, if someone with an Intel machine on macOS 11 is interested, testing the build I've linked in #7 would be very helpful, as my only modern Mac is in daily use for school, so I unfortunately don't have the time to set up a new macOS 11 test install at the moment.

**Comment by ThatsJustCheesy** [comment]
This is hopefully resolved by #10. @MrKai77, if you're still interested, please try the test build: https://github.com/WiiController/WiiController/pull/10#issuecomment-830727679

In short, I've converted the kext (deprecated) to a dext (supported and recommended) for macOS 10.15 and above, including Big Sur. Older OS versions will continue to use the kext.

Since the dext is fairly architecture-agnostic, it will hopefully _just work_ on M-series. Testing would be appreciated. I'm also getting an M1 MBA sometime next week, so I'll test it on that myself.

**Comment by ThatsJustCheesy** [comment]
I've just tested on an M1 machine and it appears to work! As usual, SIP must be disabled.

Will draft a release shortly. If you encounter any problems, please feel free to open another issue.
